plugins/mips/mips_arithmetic.ml

```diff
@@ -0,0 +1,234 @@
+open Mips.Std
+
+(* ADD rd, rs, rt
+ * Add Word, MIPS32
+ * Page 36 *)
+let add cpu ops =
+  let rd = signed cpu.reg ops.(0) in
+  let rs = signed cpu.reg ops.(1) in
+  let rt = signed cpu.reg ops.(2) in
+  RTL.[ rd := rs + rt; ]
+
+(* ADD.S fd, fs, ft , MIPS32
+ * ADD.D fd, fs, ft , MIPS32
+ * ADD.PS fd, fs, ft , MIPS32, MIPS64
+ * Floating Point Add
+ * Page 37 *)
+let add_fmt cpu ops =
+  RTL.[]
+
+(* ADDI rt, rs, immediate
+ * Add Immediate Word, MIPS32, removed in Release 6
+ * Page 38 *)
+let addi cpu ops =
+  let rt = signed cpu.reg ops.(0) in
+  let rs = signed cpu.reg ops.(1) in
+  let im = signed imm ops.(2) in
+  RTL.[ rt := rs + im; ]
+
+(* ADDIU rt, rs, immediate
+ * Add Immediate Unsigned Word, MIPS32
+ * Page 39 *)
+let addiu cpu ops =
+  let rt = unsigned cpu.reg ops.(0) in
+  let rs = unsigned cpu.reg ops.(1) in
+  let im = signed imm ops.(2) in
+  RTL.[ rt := rs + im; ]
+
+(* ADDIUPC rs, immediate
+ * Add Immediate to PC (unsigned), MIPS32 Release 6
+ * Page 40 *)
+let addiupc cpu ops =
+  let rs = unsigned cpu.reg ops.(0) in
+  let im = signed imm ops.(1) in
+  RTL.[ rs := cpu.cia + im; ]
+
+(* ADDU rd, rs, rt
+ * Add Unsigned Word, MIPS32,
+ * Page 41 *)
+let addu cpu ops =
+  let rd = unsigned cpu.reg ops.(0) in
+  let rs = unsigned cpu.reg ops.(1) in
+  let rt = unsigned cpu.reg ops.(2) in
+  RTL.[ rd := rs + rt; ]
+
+(* CLO rd, rs
+ * Count Leading Ones in Word, MIPS32
+ * Page 136 *)
+(* TODO: check that RT = RD *)
+let clo cpu ops =
+  let rs = unsigned cpu.reg ops.(0) in
+  let rt = unsigned cpu.reg ops.(1) in
```

Comment:
There are few places like that, when I have something like 
`Warning 26: unused variable rt.`
So could you please fix it by sticking to one of the following: either just remove such lines, or, if you would like to emphasize that the operand is not used, prefix it with `_` : `let _rt = `.